python/ray/dataframe/dataframe.py

```diff
@@ -192,6 +201,15 @@ def _map_partitions(self, func, index=None):
 
         return DataFrame(new_df, self.columns, index=index)
 
+    def _update_inplace(self, new_dfs):
```

Comment:
If we're adding this convenience method (which I agree is a good idea) we will also need to update `columns` and `index`.